src/Servers/HttpSys/src/RequestProcessing/Request.cs

```diff
@@ -61,43 +61,40 @@ internal Request(RequestContext requestContext, NativeRequestContext nativeReque
             var rawUrlInBytes = _nativeRequestContext.GetRawUrlInBytes();
             var originalPath = RequestUriBuilder.DecodeAndUnescapePath(rawUrlInBytes);
 
+            PathBase = string.Empty;
+            Path = originalPath;
+
             // 'OPTIONS * HTTP/1.1'
             if (KnownMethod == HttpApiTypes.HTTP_VERB.HttpVerbOPTIONS && string.Equals(RawUrl, "*", StringComparison.Ordinal))
             {
                 PathBase = string.Empty;
                 Path = string.Empty;
             }
-            else if (requestContext.Server.RequestQueue.Created)
+            else
             {
                 var prefix = requestContext.Server.Options.UrlPrefixes.GetPrefix((int)nativeRequestContext.UrlContext);
-
-                if (originalPath.Length == prefix.PathWithoutTrailingSlash.Length)
-                {
-                    // They matched exactly except for the trailing slash.
-                    PathBase = originalPath;
-                    Path = string.Empty;
-                }
-                else
+                // Prefix may be null if the requested has been transfered to our queue
```

Comment:
What will nativeRequestContext.UrlContext be in that case?

Reply:
I think this happened when using `HttpApi.DelegateRequest`. I can't repro this anymore`HttpApi.DelegateRequestEx`

Reply:
Then remove this change.

Reply:
Ah got it now. If you have multiple bindings you could get an Id that doesn't correspond to an Id in UrlPrefixes.
When that happens you end up with prefix == null and fall into the else block.

I hit this issue when my backend app had two different url prefixes and delegated to the second one.